### PR TITLE
Update emcee usage to avoid deprecation warnings

### PR DESCRIPTION
**Pull request by kboone**
Update several emcee calls that have been deprecated.

### CODE REVIEW

**Comment by kboone** [comment]
There was a code coverage failure on my last commit because we weren't testing the PTSampler branch of `mcmc_lc`. It turns out that `emcee` dropped the PTSampler code several years ago, so it doesn't work at all. Seeing as nobody ever commented about this, I don't think that anybody is using the PTSampler. I think that we should drop support for the PTSampler altogether. I kept the "sampler" argument to `mcmc_lc` for API compatibility, but it only supports ensemble now.

**Comment by kboone** [comment]
I added some tests for `mcmc_lc` since it wasn't being tested previously.

**Comment by kboone** [comment]
@benjaminrose Could you take a quick look at this before I merge it?

**Comment by benjaminrose** [comment]
We should update the dependencies to indicate that we no longer work with emcee v2.*. And I assume you want to add an option for the other moves later?

File: sncosmo/fitting.py [hunk 1]
```diff
@@ -1349,31 +1345,13 @@ def lnprior(parameters):
     def lnprob(parameters):
         return lnprior(parameters) + lnlike(parameters)
 
-    # Heuristic determination of walker initial positions: distribute
-    # walkers uniformly over parameter space. If no bounds are
-    # supplied for a given parameter, use a heuristically determined
-    # scale.
+    # Moves to use
+    moves = emcee.moves.StretchMove(a=a)
```
[the remark this answers is not in the document]
Comment (kboone) [reply sncosmo/fitting.py:1349]: That's a good point. It looks like this API was introduced in v3, and they deprecated the old one at the same time. I don't want to have to support both, so I'll update the dependency to v3.

**Comment by kboone** [comment]
I updated the minimum supported emcee version in setup.cfg. For now, let's just support the emcee default sampler. If somebody asks for it in the future we can add other moves.

**Pull request merged**